src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1247,6 +1247,15 @@ private void ImportReadOnlyPrefix()
 
         private void ImportThrow()
         {
+            var exceptionObject = _stack.Pop();
+
+            if (TrapFunction.Pointer == IntPtr.Zero)
+            {
+                TrapFunction = LLVM.AddFunction(Module, "llvm.trap", LLVM.FunctionType(LLVM.VoidType(), Array.Empty<LLVMTypeRef>(), false));
+            }
+            LLVM.BuildCall(_builder, TrapFunction, Array.Empty<LLVMValueRef>(), string.Empty);
+
+            LLVM.BuildUnreachable(_builder);
```

Comment:
Out of curiosity, was this required? My reading on llvm.trap is that LLVM already knows code afterward is unreachable.

Reply:
Yes, right now it's completely useless. I guess the same, trap would also instruct the optimizer to discard the rest of the code. I just thought that it could be useful (as a hint for me or other committers) when a real implementation for throw gets implemented.
I'm going to remove it.